labellab-client/src/reducers/auth.js

```diff
@@ -139,6 +142,32 @@ const auth = (state = initialState, action) => {
         passwordUpdatedMessage: action.payload,
         passwordUpdateError: true
       }
+      case OAUTH_LOGIN_REQUEST:
+        return {
+          ...state,
+          isAuthenticating: true
+        }
+      case OAUTH_LOGIN_SUCCESS:
+        return {
+          ...state,
+          isAuthenticated: true,
+          isAuthenticating: false,
+          statusText: 'You are logged in successfully!',
+          error: false,
+          details: {
+            email: action.payload.email
+          }
+        }
+      case OAUTH_LOGIN_FAILURE:
+        return {
+          ...state,
+          isAuthenticated: false,
+          isAuthenticating: false,
+          statusText:
+            action.error === 'Unauthorize' ? 'LOGIN FAILED!' : null,
```

Comment:
You can actually create a function for returning success message or failure message from backend.